### PR TITLE
Make Lookup instantiatable with a default constructor retaining the original behavior

### PR DESCRIPTION
**Pull request by metlos**
This should help with the testability of majority of other parts of the codebase because we can start introducing overloads accepting the new `SentryConfiguration` instance instead of statically calling `Lookup` (which implements the `SentryConfiguration` interface).

`SentryConfiguration` being an interface makes it trivially mockable without resorting to the ugliness needed to mock JNDI look up et al.

~~Note that I didn't implement any new tests yet to check the new capabilities of `Lookup` nor did I update any existing code or tests to take advantage of it. That said, `Lookup` itself uses the new code and the testsuite seems to be passing with it.~~

### CODE REVIEW

**Comment by metlos** [comment]
I've created this merely as a PR because of the lack of tests for the new functionality. Let's discuss the implementation approach in the meantime :)

File: sentry/src/main/java/io/sentry/config/Lookup.java [hunk 1]
```diff
@@ -1,90 +1,123 @@
...
+    }
 
+    private static List<ConfigurationResourceLocator> getDefaultResourceLocators() {
+        return asList(new SystemPropertiesBasedLocator(), new EnvironmentBasedLocator(), new DefaultLocator());
     }
 
-    private static String getConfigFilePath() {
-        String filePath = System.getProperty("sentry.properties.file");
+    private static List<ConfigurationProvider> getDefaultHighPriorityConfigurationProviders() {
+        boolean jndiPresent = JndiSupport.isAvailable();
```
Comment (bruno-garcia) [inline sentry/src/main/java/io/sentry/config/Lookup.java]: We just have to make sure this call won't block the android UI as per #708.

File: sentry/src/main/java/io/sentry/config/provider/PropertiesConfigurationProvider.java [hunk 2]
```diff
@@ -0,0 +1,28 @@
+package io.sentry.config.provider;
+
+import java.util.Properties;
+
+import io.sentry.util.Nullable;
+
+/**
+ * A trivial wrapper to use a {@link Properties} instance as a configuration provider.
+ */
+public class PropertiesConfigurationProvider implements ConfigurationProvider {
```
Comment (bruno-garcia) [inline sentry/src/main/java/io/sentry/config/provider/PropertiesConfigurationProvider.java]: I suggest we could live without this type.
One less abstraction cost of calling `properties == null ? null : properties.getProperty(key);` in all implementations.

If I'm not missing anything, this far we only have 1 implementation anyway.

File: sentry/src/main/java/io/sentry/config/Lookup.java (hunk 1 above)
Comment (metlos) [reply sentry/src/main/java/io/sentry/config/Lookup.java]: Not sure how to test that atm. All I know is that the behavior is the same as it was with the prior implementation of `Lookup` which did the exact same check as is done in `JndiSupport.isAvailable()` during the first config property lookup. This is exactly the behavior now as well, where we instantiate the inner "deprecated" instance of Lookup on the first config lookup lazily.

**Comment by metlos** [comment]
Note that I've started using Mockito in the new tests, for better or worse. IMHO, it is a de-facto industry standard and doesn't allow for the insanity of mocking system classes like jmockit does. IMHO, that's for good reasons.

**Pull request merged**